lib/redux/actions/async.js

```diff
@@ -557,6 +625,17 @@ export function setUploadTargetUserAndMaybeRedirect(targetId) {
     });
     if (_.isEmpty(targetedDevices) || !atLeastOneDeviceSupportedOnSystem) {
       return dispatch(syncActions.setPage(pages.SETTINGS));
+    } else {
+      if (actionUtils.userHasRole(_.get(allUsers, loggedInUser), 'clinic')) {
```

Comment:
Similar comment as above for `clickEditUserNext` but also see comment about not reusing this action creator in the clinic case...I think we could simplify by having something separate, especially as I think also the `setBlipViewDataUrl` part is unnecessary? Am I right that there's no "Go to Blip" button in the clinic flow?